examples/run_language_modeling.py

```diff
@@ -193,6 +207,16 @@ def main():
             "You are instantiating a new tokenizer from scratch. This is not supported, but you can do it from another script, save it,"
             "and load it from here, using --tokenizer_name"
         )
+    if tokenizer.pad_token_id is None:
+        if model_args.force_pad_token:
+            # See PR 3388. Some tokenizers don't had pad tokens which causes errors at the encoding step in the collate_fn.
+            # We give here the option to force the addition of a pad token. The attention mask is used to ignore this token
+            # when feeding to the model.
+            tokenizer.add_special_tokens({"pad_token": "<pad>"})
```

Comment:
I don't think this will work since it will give the `pad_token` the id: `len(tokenzier) + 1` that does not exist in the model embedding weights. What one could do for GPT2 is to set the pad_token_id = eos_token => `tokenizer.pad_token = tokenizer.eos_token`. Since GPT2 uses causal masking, this should be fine.

Reply:
@patrickvonplaten I think it should work since later on the embeddings are resized:

https://github.com/huggingface/transformers/blob/5ff6eb7073fc6959b4cd5464defddc101d3163f8/examples/run_language_modeling.py#L232

Reply:
Ah yeah you're right - I didn't even realize that there is resize embedding in the script as well. 

I don't think though one should add a new embedding weight for GPT2, but just reuse some other token for padding so that not the whole model has to be retrained. `model.resize_token_embeddings` only adds new tokens if `len(tokenizer)` is > then `model.old_embedding_weights`. A lot of people just wanting to fine-tune GPT2 can run into bad performance here without knowing why.

Reply:
Thinking a bit more about it, I agree with @julien-c and think people should just adapt this script for their own (GPT2) need. It's really not that long, and providing hacky functionality here is not worth it.

Reply:
And the script does work for GPT2 by default, i.e. if you don't opt in to `--line_by_line`